rpcs3/Emu/IdManager.h

```diff
@@ -27,10 +27,10 @@ namespace id_manager
 	template <typename T>
 	struct id_traits<T, std::void_t<decltype(&T::id_base), decltype(&T::id_step), decltype(&T::id_count)>>
 	{
-		static const u32 base    = T::id_base;
-		static const u32 step    = T::id_step;
-		static const u32 count   = T::id_count;
-		static const u32 invalid = base > 0 ? 0 : -1;
+		static const u32 base        = T::id_base;
+		static const u32 step        = T::id_step;
+		static const u32 count       = T::id_count;
+		static constexpr u32 invalid = base > 0 ? 0 : -1;
```

Comment:
How about
```suggestion
		static const u32 invalid = -!base;
```

That's really interesting...

Reply:
Atleast -u32{!base} to avoid warning.

Reply:
`-!` doesnt seem to cause any warnings.

Reply:
I would assume the default type path of `u32 = -int(bool(!base))` produces a warning on some compilers, have you checked all compilers?

Reply:
It triggers `warning C4804:  '-': unsafe use of type 'bool' in operation` for anyone using VS solution files. For cmake this warning has been disabled in ConfigureCompiler.cmake.

Reply:
Well, `-+!base` then. Who can do shorter.